tests/bases/test_metric.py

```diff
@@ -362,3 +364,42 @@ def device(self):
     assert module.device == module.metric.device
     if isinstance(module.metric.x, Tensor):
         assert module.device == module.metric.x.device
+
+
+@pytest.mark.parametrize("device", ["cpu", "cuda"])
+@pytest.mark.parametrize("requires_grad", [True, False])
+def test_constant_memory(device, requires_grad):
+    """Checks that when updating a metric the memory does not increase."""
+    if not torch.cuda.is_available() and device == "cuda":
+        pytest.skip("Test requires GPU support")
+
+    def get_memory_usage():
```

Comment:
Curious if this is safe in case you run tests in parallel?

Reply:
I think it is safe because we explicit ask for the memory consumption of the current process